### ISSUE TITLE
Add a Teacher view

### ISSUE THREAD

**Issue by PetalCat** [PSDTools/app#192]

**Comment by lishaduck** [comment]
Yep! The current system is split up into lots of small, modular widgets so that they can be on separate pages.

**Comment by lishaduck** [comment]
You wanna tackle this next after #193, @ParkerH27?

**Comment by lishaduck** [comment]
Made a few changes to the surrounding code to make this a easier.

**Comment by PetalCat** [comment]
> You wanna tackle this next after #193, @ParkerH27?

I guess so.... #193 is brain breaking

**Comment by lishaduck** [comment]
> > You wanna tackle this next after #193, @ParkerH27?
> 
> I guess so.... #193 is brain breaking

Yeah. Flaky tests are no fun. Did you look at 57339e90d313f7fb0ad77db5eb6451745276de19? I've found [invertase](https://invertase.io)'s [_Assertions In Dart And Flutter Tests: An (Now For Sure) Ultimate Cheat Sheet_](https://invertase.io/blog/assertions-in-dart-and-flutter-tests-now-for-sure-an-ultimate-cheat-sheet) to be quite useful. Also [_Flaky Tests: Getting Rid Of A Living Nightmare In Testing_](https://www.smashingmagazine.com/2021/04/flaky-tests-living-nightmare/) is a good article on why tests flake.

**Comment by lishaduck** [comment]
It's been 2 months. You still up for this? We need to get it done before we present (b/c it's blocking me on some other stuff).

**Comment by ParkerH27** [comment]
A: Hopefully that is what I was supposed to do because it felt waaayyy too easy(that is scary)
B: You're welcome for somehow failing the Markdown test :)

**Comment by lishaduck** [comment]
> A: Hopefully that is what I was supposed to do because it felt waaayyy too easy(that is scary)
> B: You're welcome for somehow failing the Markdown test :)

I'll address both of these in turn.

- B: It's a bug, something changed on GitHub's side. I've removed the auto-pr to fix markdownlint failures, so it works now.
- A: It's a good start. However, I'd personally recommend starting over. You moved all the code all over the place, when really all you needed to do was remove `currentStageProvider` and the `Stage` class. Then, rename `_StudentView` and `_TeacherView` to lack an underscore, annotate them with `@RoutePage()` as you did, and split them into two files. At that point, see what's erroring, and switch them to use routes instead of Riverpod. It should also be noted that the tests are failing because of how you configured the router. Retain the current setup, but have the main page be like `WrapperPage`, having children in the `AppRouter` singleton and using the `AutoRouter` widget to get children. Maybe use the `guards` list to bounce people to the right page? Another thing to keep in mind is that you can use [query parameters](https://github.com/Milad-Akarie/auto_route_library?tab=readme-ov-file#query-parameters) for `ViewCoinsStage.student`, which will let you make that passable in navigation.
